cmdstanpy/install_cmdstan.py

```diff
@@ -150,51 +143,77 @@ def build(verbose: bool = False) -> None:
         )
 
 
-def compile_example() -> None:
+# pylint: disable=no-self-use
+def _wrap_build_progress_hook(
+    cmdstan_version: str,
+) -> Optional[Callable[[str], None]]:
+    """Sets up tqdm callback for CmdStan sampler console msgs."""
+    pad = ' ' * 20
+    pbar: Any = tqdm(
+        total=200,
+        bar_format="{desc} ({elapsed}) | {bar} | {postfix[0][value]}",
+        postfix=[dict(value=f'Building {cmdstan_version} {pad}')],
+        colour='blue',
+        desc='',
+        position=0,
+    )
+    success = f'--- CmdStan {cmdstan_version} built ---'
+
+    def build_progress_hook(line: str) -> None:
+        if line == '__DONE__' or line == success:
```

Comment:
Could we do this with a simple regex instead to make this function version-agnostic and resolve the rebuild issue? We lose the ability to postfix the version we're building but I think its fine if the bar just says "Building Cmdstan" instead of "Building 2.27.0"

Reply:
Alt: We can refactor `utils.cmdstan_version_at` into two functions, one which parses the makefile to get the version and the other that has the current functionality, and then use the first in `build`. I think I prefer the regex/fuzzy solution, but either seems fine to me.

Reply:
I simplified the condition to eliminate the need for the regex, also, changed "do_command" to send final line of stdout to pbar (as needed) and it seems to just work and is in fact, the right thing to do.